cmd/dfget/app/root.go

```diff
@@ -0,0 +1,217 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	cfg "github.com/dragonflyoss/Dragonfly/dfget/config"
+	"github.com/dragonflyoss/Dragonfly/dfget/core"
+	"github.com/dragonflyoss/Dragonfly/dfget/errors"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	localLimit string
+	totalLimit string
+	filter     string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "dfget",
+	Short: "The dfget is the client of Dragonfly.",
+	Long:  "The dfget is the client of Dragonfly, a non-interactive P2P downloader.",
+	Run: func(cmd *cobra.Command, args []string) {
+		checkParameters()
+		initLog()
+		cfg.AssertContext(cfg.Ctx)
+		cfg.Ctx.ClientLogger.Infof("cmd params:%q", args)
+		initProperties()
+		cfg.Ctx.ClientLogger.Infof("context:%s", cfg.Ctx)
+		err := core.Start(cfg.Ctx)
+		if err != nil {
+			util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+			os.Exit(err.Code)
+		}
+		util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+		os.Exit(0)
+	},
+}
+
+func init() {
+	initFlags()
+}
+
+func checkParameters() {
+	fmt.Println(os.Args)
+	if len(os.Args) < 3 {
+		fmt.Println("Please use '--help' or '-h' to show the help information.")
+		os.Exit(0)
+	}
+}
+
+func resultMsg(ctx *cfg.Context, end time.Time, e *errors.DFGetError) string {
+	if e != nil {
+		return fmt.Sprintf("download FAIL(%d) cost:%.3fs length:%d reason:%d error:%v",
+			e.Code, end.Sub(ctx.StartTime).Seconds(), ctx.RV.FileLength,
+			ctx.BackSourceReason, e)
+	}
+	return fmt.Sprintf("download SUCCESS(0) cost:%.3fs length:%d reason:%d",
+		end.Sub(ctx.StartTime).Seconds(), ctx.RV.FileLength, ctx.BackSourceReason)
+}
+
+// initProperties
+func initProperties() {
+	for _, v := range cfg.Ctx.ConfigFiles {
+		if err := cfg.Props.Load(v); err == nil {
+			cfg.Ctx.ClientLogger.Debugf("initProperties[%s] success: %v", v, cfg.Props)
+			break
+		} else {
+			cfg.Ctx.ClientLogger.Debugf("initProperties[%s] fail: %v", v, err)
+		}
+	}
+
+	if cfg.Ctx.Node == nil {
+		cfg.Ctx.Node = cfg.Props.Nodes
+	}
+
+	if cfg.Ctx.LocalLimit == 0 {
+		cfg.Ctx.LocalLimit = cfg.Props.LocalLimit
+	}
+
+	if cfg.Ctx.TotalLimit == 0 {
+		cfg.Ctx.TotalLimit = cfg.Props.TotalLimit
+	}
+
+	if cfg.Ctx.ClientQueueSize == 0 {
+		cfg.Ctx.ClientQueueSize = cfg.Props.ClientQueueSize
+	}
+
+	cfg.Ctx.Filter = transFilter(filter)
+
+	// be compatible with dfget python version
```

Comment:
Is there necessary to compatible with python version? @Starnop

Reply:
@lowzj  Can we make a unity here? In my opinion, a string type might be more friendly for users. WDYT?

Reply:
Sure